survey/survey_app/serializers.py

```diff
@@ -30,19 +33,52 @@ class Meta:
 
 
 class AnswerSerializer(serializers.ModelSerializer):
+    # Add the 'required=True' parameter to the 'question' field
+    question = serializers.PrimaryKeyRelatedField(queryset=Question.objects.all(), required=True)
+    option = serializers.PrimaryKeyRelatedField(queryset=Option.objects.all())
+
     class Meta:
         model = Answer
-        fields = '__all__'
+        fields = ['question', 'option']
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        if user:
```

Comment:
We offen check `user.is_authenticated` rather than just `user` @newDaria

Reply:
updated